sw/airborne/modules/guidance/gvf/nav/nav_survey_polygon_gvf.c

```diff
@@ -0,0 +1,326 @@
+}
+
+/**
+ *  initializes the variables needed for the survey to start
+ *  @param first_wp      the first Waypoint of the polygon
+ *  @param size          the number of points that make up the polygon
+ *  @param angle         angle in which to do the flyovers
+ *  @param sweep_width   distance between the sweeps
+ *  @param shot_dist     distance between the shots
+ *  @param min_rad       minimal radius when navigating
+ *  @param altitude      the altitude that must be reached before the flyover starts
+ **/
+void gvf_nav_survey_polygon_setup(uint8_t first_wp, uint8_t size, float angle, float sweep_width, float shot_dist,
+                              float min_rad, float altitude)
+{
+  int i;
+  struct FloatVect2 small, sweep;
+  float divider, angle_rad = angle / 180.0 * M_PI;
+
+  if (angle < 0.0) { angle += 360.0; }
+  if (angle >= 360.0) { angle -= 360.0; }
+
+  gvf_survey.poly_first = first_wp;
+  gvf_survey.poly_count = size;
+
+  gvf_survey.psa_sweep_width = sweep_width;
+  gvf_survey.psa_min_rad = min_rad;
+  gvf_survey.psa_shot_dist = shot_dist;
+  gvf_survey.psa_altitude = altitude;
+
+  gvf_survey.segment_angle = angle;
+  gvf_survey.return_angle = angle + 180;
+  if (gvf_survey.return_angle > 359) { gvf_survey.return_angle -= 360; }
+
+  if (angle <= 45.0 || angle >= 315.0) {
+    //north
+    gvf_survey.dir_vec.y = 1.0;
+    gvf_survey.dir_vec.x = 1.0 * tanf(angle_rad);
+    sweep.x = 1.0;
+    sweep.y = - gvf_survey.dir_vec.x / gvf_survey.dir_vec.y;
+  } else if (angle <= 135.0) {
+    //east
+    gvf_survey.dir_vec.x = 1.0;
+    gvf_survey.dir_vec.y = 1.0 / tanf(angle_rad);
+    sweep.y = - 1.0;
+    sweep.x = gvf_survey.dir_vec.y / gvf_survey.dir_vec.x;
+  } else if (angle <= 225.0) {
+    //south
+    gvf_survey.dir_vec.y = -1.0;
+    gvf_survey.dir_vec.x = -1.0 * tanf(angle_rad);
+    sweep.x = -1.0;
+    sweep.y = gvf_survey.dir_vec.x / gvf_survey.dir_vec.y;
+  } else {
+    //west
+    gvf_survey.dir_vec.x = -1.0;
+    gvf_survey.dir_vec.y = -1.0 / tanf(angle_rad);
+    sweep.y = 1.0;
+    sweep.x = - gvf_survey.dir_vec.y / gvf_survey.dir_vec.x;
+  }
+
+  //normalize
+  FLOAT_VECT2_NORMALIZE(sweep);
+
+  VECT2_SMUL(gvf_survey.rad_vec, sweep, gvf_survey.psa_min_rad);
+  VECT2_SMUL(gvf_survey.sweep_vec, sweep, gvf_survey.psa_sweep_width);
+
+  //begin at leftmost position (relative to gvf_survey.dir_vec)
+  VECT2_COPY(small, waypoints[gvf_survey.poly_first]);
+
+  divider = (gvf_survey.sweep_vec.y * gvf_survey.dir_vec.x) - (gvf_survey.sweep_vec.x * gvf_survey.dir_vec.y);
+
+  //calculate the leftmost point if one sees the dir vec as going "up" and the sweep vec as going right
+  if (divider < 0.0) {
+    for (i = 1; i < gvf_survey.poly_count; i++)
+      if ((gvf_survey.dir_vec.x * (waypoints[gvf_survey.poly_first + i].y - small.y)) + (gvf_survey.dir_vec.y *
+          (small.x - waypoints[gvf_survey.poly_first + i].x)) > 0.0) {
+        VECT2_COPY(small, waypoints[gvf_survey.poly_first + i]);
+      }
+  } else
```

Comment:
please add brackets to the else clause (or run the fix_code_style script over the file)